lib/proxy/client_test.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2022 Gravitational, Inc
```

Comment:
It looks like all the tests here are exercising private functions. It would be nice to see a few integration tests that have Proxies/Nodes and actually call `Dial`

Reply:
I have this template that tests the old behaviour here https://github.com/gravitational/teleport/pull/11671
Once all the changes are in it should be as simple as running a pretty similar test with the new config.